examples/docker/config.machines.yml

```
@@ -1,404 +1,49 @@
       stay_connected: !!bool true
-      collectors:
+      collectors: # Automatically apply transports
       - l99.driver.fanuc.collectors.MachineInfo, fanuc
       - l99.driver.fanuc.collectors.Alarms, fanuc
       - l99.driver.fanuc.collectors.Messages, fanuc
       - l99.driver.fanuc.collectors.StateData, fanuc
       - l99.driver.fanuc.collectors.ToolData, fanuc
       - l99.driver.fanuc.collectors.ProductionData, fanuc
-      #- l99.driver.fanuc.collectors.ProductionDataExternalSubprogramDetails, fanuc
+      - l99.driver.fanuc.collectors.ProductionDataExternalSubprogramDetails, fanuc
       - l99.driver.fanuc.collectors.GCodeData, fanuc
       - l99.driver.fanuc.collectors.AxisData, fanuc
       - l99.driver.fanuc.collectors.SpindleData, fanuc
-    l99.driver.fanuc.transports.SpB, fanuc:
-      overrides:
-        scada_id: ~
-        group_id: ~
-        node_id: ~
-        device_id: ~
-      net:
-        type: tcp
-        ip: 10.1.10.2
-        port: !!int 1883
-      anonymous: !!bool false
-      user: admin
-      password: password
-      reconnect_s: !!int 30
+      - l99.driver.fanuc.collectors.Macro, fanuc
+      - l99.driver.fanuc.collectors.Pmc, fanuc
 
-    ##
-    ## influxdb example
-    ##
-  - id: f_sim_influx_long
-    enabled: !!bool false
-    type: l99.driver.fanuc.FanucMachine, fanuc
-    strategy: l99.driver.fanuc.strategies.FanucMultiStrategy, fanuc
-    handler: l99.driver.fanuc.handlers.FanucOne, fanuc
-    transport: l99.driver.fanuc.transports.InfluxLP, fanuc
-    l99.driver.fanuc.FanucMachine, fanuc:
-      sweep_ms: !!int 1000
+    l99.driver.fanuc.handlers.FanucOne, fanuc:
+      iso8601: !!bool true
+      change_only: !!bool true
+      skip_internal: !!bool true
+
+    l99.driver.fanuc.transports.MQTT, fanuc:
+      publish_model: !!bool false ## enable this to metadata about machine configuration
+      topic: fanuc/{{machine.Id}}/{{veneer.Name}}{{if veneer.SliceKey}}/{{veneer.SliceKey}}{{end}}
       net:
-        ip: 192.168.111.12
-        port: !!int 8193
-        timeout_s: !!int 3
-    l99.driver.fanuc.strategies.FanucMultiStrategy, fanuc:
-      stay_connected: !!bool false
-      collectors:
-      - l99.driver.fanuc.collectors.MachineInfo, fanuc
-      - l99.driver.fanuc.collectors.Alarms, fanuc
-      - l99.driver.fanuc.collectors.Messages, fanuc
-      - l99.driver.fanuc.collectors.StateData, fanuc
-      - l99.driver.fanuc.collectors.ToolData, fanuc
-      - l99.driver.fanuc.collectors.ProductionData, fanuc
-      - l99.driver.fanuc.collectors.GCodeData, fanuc
-      - l99.driver.fanuc.collectors.AxisData, fanuc
-      - l99.driver.fanuc.collectors.SpindleData, fanuc
-    l99.driver.fanuc.transports.InfluxLP, fanuc:
-      <<: *default-influx-transformers
-      host: "http://influx:8086"
-      token: "dewlnpy0CmcikwWmMBlqoxcoeuUWpdlDQi8yH7QyGU0tcfn0avMMcpPWYJ-ZUmLD8iHXNl27M3GBHGC4n4_EwA=="
-      org: "fanuc"
-      bucket: "fanuc"
+        type: tls
+        ip: localhost
```

If you want to access mqtt server from outside the local computer, you will need to change it to this
```suggestion
        ip: 0.0.0.0
```